### PR TITLE
Make build the build process much faster

### PR DESCRIPTION
**Pull request by mlorbetske**
TODO:
- ~~Add back downloading the CLI~~
- The actual (not PR) CI agent will need to be updated to point at the new locations
- ~~Update the script on the share to account for Loc.Settings.targets moving in to the build folder~~

This also makes it possible to do `dotnet new3` at the repo root again

### CODE REVIEW

**Comment by mlorbetske** [comment]
I think the next real step here is to merge to stabilize to see if we're getting good builds

**Comment by mlorbetske** [comment]
Looks like JSON.NET versions may be getting set incorrectly, there’s at least one instance where 8.0.3 shows up in Linux builds

File: build/CoreBuild.proj [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    <PropertyGroup>
+        <RepoRoot>$(MSBuildThisFileDirectory)..\</RepoRoot>
+        <ArtifactsDir>$(RepoRoot)artifacts\</ArtifactsDir>
+        <DevPath>$(RepoRoot)dev\</DevPath>
+        <_PackageOutputPath>$(ArtifactsDir)packages\</_PackageOutputPath>
+        <TestResultsPath>$(ArtifactsDir)test_results\</TestResultsPath>
+        <TemplatesOutputPath>$(ArtifactsDir)templates\</TemplatesOutputPath>
+        <CopyBuildOutputToOutputDirectory>false</CopyBuildOutputToOutputDirectory>
+        <IncludeBuildOutput>False</IncludeBuildOutput>
+        <IncludeBuildOutput>False</IncludeBuildOutput>
```
Comment (seancpeters) [inline build/CoreBuild.proj]: Duplicate line

**Comment by mlorbetske** [comment]
Running a test build on the real CI

**Pull request merged**